app/services/hyrax/collections/collection_types_service.rb

```diff
@@ -0,0 +1,49 @@
+module Hyrax
+  module Collections
+    module CollectionTypesService
+      # @api public
+      #
+      # What types of collection can the user create or manage
+      #
+      # @param user [User] user - The user requesting to create/manage a Collection
+      # @param roles [String] type of access, 'manage' and/or 'create'
+      # @return [Array<Hyrax::CollectionType>]
+      def self.collection_types_for_user(user, roles)
```

Comment:
Would you be able to switch to using named parameters?

```ruby
self.collection_types_for_user(user:, roles:)
```

Then callers wouldn't need to remember the parameter order when calling.

(This is a non-blocker)

Reply:
I was envisioning this method a bit differently.  The service that is needed is one that returns collection types that can be created by a user.  Perhaps this would be an additional class method in this service class.  I was thinking something like this...

```
      # @api public
      #
      # Get a list of collection types that a user can create
      #
      # @param user [User] the user that will be creating a collection (default: current_user)
      # @return [Array<Hyrax::CollectionType>] array of collection types the user can create
      def self.can_create_collection_types(user=current_user)
        # Stubbed to return all types.
        return Hyrax::CollectionType.all if user.present? # condition test added to allow paramter to be present
        # TODO: implement the following per issue #1582
        # all collection types when user is an admin (i.e., in the admin group)
        # collection types where user has manage access for collection types by being a user with manage access
        # colleciton types where user has manage access by being in a group with manage access
        # collection types where user has create access for collections types by being a user with create access
        # collection types where user has create access by being in a group with create access
        []
      end
```

Reply:
Both these requests are addressed now @elrayle and @jeremyf, if you want to take a look.